tidy3d/web/api/tidy3d_stub.py

```diff
@@ -0,0 +1,87 @@
+"""Stub for webapi"""
+from __future__ import annotations
+
+from typing import Union
+
+from pydantic.v1 import BaseModel
+
+from tidy3d import SimulationData, ModeSolverData
+from tidy3d.web.core.types import TaskType
+from tidy3d.components.simulation import Simulation
+from tidy3d.plugins.mode.mode_solver import ModeSolver
+
+
+class Tidy3dStub(BaseModel):
+
+    component: Union[Simulation, ModeSolver]
+
+    @classmethod
+    def from_file(cls, file_path: str) -> Union[Simulation, ModeSolver]:
+        """Loads a :class:`Stub` from .yaml, .json, or .hdf5 file.
+
+        Parameters
+        ----------
+        file_path : str
+            Full path to the .yaml or .json or .hdf5 file to load the :class:`Stub` from.
+
+        Returns
+        -------
+        :class:`Stub`
+            An instance of the component class calling `load`.
+
+        Example
+        -------
+        >>> simulation = Simulation.from_file(fname='folder/sim.json') # doctest: +SKIP
+        """
+        pass
+
+    def to_file(
+        self,
+        file_path: str,
+    ):
+        """Exports :class:`Stub` instance to .yaml, .json, or .hdf5 file
+
+        Parameters
+        ----------
+        file_path : str
+            Full path to the .yaml or .json or .hdf5 file to save the :class:`Stub` to.
+
+        Example
+        -------
+        >>> simulation.to_file(fname='folder/sim.json') # doctest: +SKIP
+        """
+        pass
+
+    def get_type(self):
+        """Get simulation instance type.
+
+        Returns
+        -------
+        :class:`TaskType`
+            An instance Type of the component class calling `load`.
+        """
+        if isinstance(self.component, Simulation):
+            return TaskType.FDTD
+        elif isinstance(self.component, ModeSolver):
+            return TaskType.MODE_SOLVER
+
+    def validate_pre_upload(self, source_required):
+        """Perform some pre-checks on instances of component"""
+        pass
+
+
+class Tidy3dStubData(BaseModel):
+    """"""
+
+    data: Union[SimulationData, ModeSolverData]
+
+    @classmethod
+    def postprocess(cls, task_type: TaskType, path: str) -> Tidy3dStubData:
```

Comment:
my idea here was rather that we provide a separate `from_file()` method and then call `self.postprocess()` 
so the generic "load" webapi would look something like
```
def load(fname):
    obj = StubData.from_file(fname)
    obj = obj.postprocess()

```